charts/aws-config-generation/templates/deployment.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "aws-oidc.fullname" . }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "aws-oidc.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "aws-oidc.selectorLabels" . | nindent 6 }}
+{{- if .Values.strategy }}
+  {{- with .Values.strategy }}
+  strategy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+{{- end }}
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "aws-oidc.selectorLabels" . | nindent 8 }}
+      namespace: {{ .Values.namespace }}
+    spec:
+      {{- range .Values.imagePullSecrets }}
+      imagePullSecrets:
+      - name: {{ . }}
+      {{- end }}
+      serviceAccount: {{ include "aws-oidc.serviceAccountName" . }}
+      serviceAccountName: {{ include "aws-oidc.serviceAccountName" . }}
+      {{- if .Values.automountServiceAccountToken }}
+      automountServiceAccountToken: {{ .Values.automountServiceAccountToken }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      dnsPolicy: {{ .Values.dnsPolicy }}
+      restartPolicy: {{ .Values.restartPolicy }}
+      schedulerName: {{ .Values.schedulerName }}
+      containers:
+        - name: {{ .Chart.Name }}
+          {{- with .Values.image.args }}
+          args:
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          env:
+          {{- range .Values.image.envs }}
+          - name: {{ .name }}
+            value: {{ .value }}
+          {{- end }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: Always
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /health
+              port: http
+              scheme: HTTP
+            failureThreshold: 3
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: http
+              scheme: HTTP
+            failureThreshold: 3
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
```

Comment:
Can we make the liveness probe optional, and make the thresholds for both probes configurable?

Reply:
we want liveness probe to be optional?

Reply:
ok I guess it makes sense for a generic template

Reply:
Keep in mind a failing liveness probe will *restart the container*

There are many reasons an application can become unresponsive - poor thread/lock management, resource starvation, and failures in *upstream services.* Liveness probes are great for restarting an application that can't resolve an internal deadlock, but this behavior can cause cascading failures when a cluster is resource constrained. Some teams choose to use liveness probes sparingly, and usually with higher thresholds than readiness probes.